operator/text/v0/convert.go

```diff
@@ -71,3 +69,75 @@ func convertToText(input ConvertToTextInput) (ConvertToTextOutput, error) {
 		Error: res.Error,
 	}, nil
 }
+
+type markdownConverter struct{}
+
+func extractMetadata(content string) map[string]string {
+	var metadata map[string]string
+
+	if !strings.HasPrefix(content, "---") {
+		return metadata
+	}
+
+	parts := strings.SplitN(content, "---", 3)
+	if len(parts) < 3 {
+		return metadata
+	}
+
+	yamlContent := parts[1]
+	err := yaml.Unmarshal([]byte(yamlContent), &metadata)
+	if err != nil {
+		return metadata
+	}
+
+	return metadata
+}
+
+func (m markdownConverter) convert(contentType string, b []byte) (ConvertToTextOutput, error) {
+
+	before := time.Now()
+	content := string(b)
+	metadata := extractMetadata(content)
+
+	duration := time.Since(before)
+	millis := duration.Milliseconds()
+
+	if metadata == nil {
+		metadata = map[string]string{}
+	}
+
+	return ConvertToTextOutput{
+		Body:  content,
+		Meta:  metadata,
+		MSecs: uint32(millis),
+		Error: "",
+	}, nil
+}
+
+func convertToText(input ConvertToTextInput) (ConvertToTextOutput, error) {
+
+	contentType, err := getContentTypeFromBase64(input.Doc)
+	if err != nil {
+		return ConvertToTextOutput{}, err
+	}
+
+	b, err := base64.StdEncoding.DecodeString(base.TrimBase64Mime(input.Doc))
+	if err != nil {
+		return ConvertToTextOutput{}, err
+	}
+
+	var converter converter
+	switch contentType {
+	case "application/octet-stream":
```

Comment:
`docconv` provides a function `MimeTypeByExtension()` that can let you know the supported MIME types in the library. 
So the flow should be:
1. Use `MimeTypeByExtension()` to check if `docconv` supports the type. If not, `MimeTypeByExtension()` will return `"application/octet-stream"`.
2. If the type is supported, we use `docconv` to convert the file.
3. If not (it returns `"application/octet-stream"`), we check if it is a text file. If it is, we directly convert it to a string. We don't need to consider whether it is markdown or not.

Reply:
@donch1989 
Thank you for telling me those!
I want to confirm a few todo points.

1. About using `MimeTypeByExtension()`
In `ConvertToTextInput.Doc`, it seems we do not have to use this function because the encoded string contains the MIME Type. And previous function `getContentTypeFromBase64(base64String string) (string, error)` can do it.
So, we do not have to call `MimeTypeByExtension()`. 

2. About judging if it is a text file
I searched for a while, but it seems that we can only judge it from [MIME types](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types).
So, do I directly check if the MIME type is text file?
e.g. application/x-abiword is not supported in `docconv`, so I add this type in the our code.

3. About metadata
In Markdown, we can extract metadata through "---" "---".
But, I couldn't find other MIME types metadata such as csv.
Do we just set it as a blank map?

Reply:
1. Yeah, you are right. I looked at it wrong. But we still need to check the MIME type listed in `MimeTypeByExtension()` or `Convert()` to know which types are supported in `docconv`.

2. I haven't try it yet, but please check [this](https://pkg.go.dev/unicode/utf8#ValidString) to see if it can work.

3. Actually, I think we don't even need to extract metadata from Markdown. The "---" "---" is just a convention, not a standard. We can just leave the metadata empty. To be more precise, the "---" "---" blocks are more like a header. They are still part of the data, not metadata.

Reply:
Markdown is just a text file with a certain pattern. It is difficult to define which part is metadata in the text file. I'd suggest we don't do this additional interpretation.

Reply:
@donch1989 
Sync
About 2., 
By ValidString, the encoded string will be true with all types of format.
So, I will use [this function](https://pkg.go.dev/unicode/utf8#Valid) to confirm if a file is a text file.

Reply:
Sync
I test the same content for test.csv & test.xlsx.
And, it turns out the result are different from the [Valid function](https://pkg.go.dev/unicode/utf8#Valid).
So, it seems that we cannot convert the xslx file by the current way.
I will research how to convert it with another way.

For the Excel file (`test.xlsx`):
The file content is not valid UTF-8. In fact, most Office files are binary files 
that don't contain text data in UTF-8 format. Therefore, `utf8.Valid(b)` returns 
false as expected.